### ISSUE TITLE
Question: How to set which environment should be used for migrations in CLI for EF Core 2.0

### ISSUE THREAD

**Issue by ivankarpey** [dotnet/efcore#8786]
I'm playing with EF Core 2.0 and have 3 env in my app: dev, staging and production. So I'm generate models, create migrations and now want to apply them to specific database. However I faced with a problem. When I executing `dotnet ef database drop (update)` for example, it's trying to get production env by default.

How could I specify what environment should be used? My DbContextFactory looks like that: 

    public class MyDbContextFactory : IDbContextFactory<MyDbContext>
        {
            public MyDbContext Create(string[] args) => //how to pass something to this args?
                WebHost.CreateDefaultBuilder(args)
                .UseStartup<Startup>()
                .Build().Services.GetRequiredService<MyDbContext>();
        }

it was something like: 
`dotnet ef database drop -e development` in version 1, but this was removed from version 2. Command like
 
```
set ASPNETCORE_ENVIRONMENT=development 
dotnet ef database drop
```

 also seems not working. In every case it's handled as Production for me. 

os - Windows 7 64 bit. 
ide - VS Code
.NET Core 2.0 preview1-final
EF Core 2.0 preview1-final

**Comment by bricelam** [comment]
```cmd
SET ASPNETCORE_ENVIRONMENT=Development
```
This should work.

Are you in a PowerShell session? If so, you'll need to use the PowerShell syntax for setting environment variables.
```ps1
$env:ASPNETCORE_ENVIRONMENT = 'Development'
```

**Comment by bricelam** [comment]
(Also [asked on Stack Overflow](https://stackoverflow.com/q/44442349/475031))

**Comment by ivankarpey** [comment]
Hmm, looks like it's work when execute from powershell (with $env:... ), but neither `set` nor `setx` from terminal not making any effect. Maybe some issue on my side. However, when running with  "development" it's fails with the error described [there](https://github.com/aspnet/Identity/issues/1239) just FYI, maybe you connected with Identity teams somehow.

**Comment by bricelam** [comment]
The *cannot resolve scoped service* error will be fixed in 2.0.0-preview2-final. You can try [the nightly builds](https://dotnet.myget.org/gallery/aspnetcore-dev) if you're eager to test it.
